app/auth.go

```diff
@@ -82,19 +146,225 @@ func NewAuthManager(conf AuthManagerConfig) AuthManager {
 	return mgr
 }
 
-func (m *MenderAuthManager) IsAuthorized() bool {
-	adata, err := m.AuthToken()
+// GetInMessageChan returns the channel to send requests to the auth manager
+func (m *MenderAuthManager) GetInMessageChan() chan<- *AuthManagerRequest {
+	return m.inChan
+}
+
+// GetAckMessageChan returns the channel to get acks from the auth manager
+func (m *MenderAuthManager) GetAckMessageChan(name string) <-chan *AuthManagerAck {
+	if m.ackChans[name] == nil {
+		m.ackChans[name] = make(chan *AuthManagerAck, AuthManagerChannelBuffer)
+	}
+	return m.ackChans[name]
+}
+
+// GetOutMessageChan returns the channel to get responses from the auth manager
+func (m *MenderAuthManager) GetOutMessageChan(name string) <-chan *AuthManagerResponse {
+	if m.outChans[name] == nil {
+		m.outChans[name] = make(chan *AuthManagerResponse, AuthManagerChannelBuffer)
+	}
+	return m.outChans[name]
+}
+
+// Run is the main routine of the Mender authorization manager
+func (m *MenderAuthManager) Run() error {
+	m.running = true
+	for m.running {
+		select {
+		case msg, ok := <-m.inChan:
+			if !ok {
+				break
+			}
+			switch msg.Method {
+			case MethodGetAuthToken:
+				// received GET_AUTH_TOKENß
+				m.ackRequest(msg.Method, msg.Sender)
+				m.getAuthToken()
```

Comment:
We can remove the ack here, as it is redundant. My idea was to keep the pattern consistent (request -> ack, then get the response in the out channel). This is because the response will be managed by `AuthManagerResponseConsumer` running in a separate goroutine, indeed.

Reply:
I removed the ack, the response message is enough, you are right.